File: NirvanaSMS.Api.CSharp/Core/Base/MessageBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NirvanaSMS.Api.Core.Base
{
    public abstract class MessageBase
    {
        public string Header { get; set; }
        public DateTime? SendDateTime { get; set; } = null;
        public int ValidityTime { get; set; }
    }
}

File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/core/base/MessageBase.java
package org.nirvanasms.api.core.base;

import java.util.Date;

public abstract class MessageBase {
	private String header;
	private Date sendDateTime;
	private int validityTime;
	
	public String getHeader() {
		return header;
	}
	public void setHeader(String header) {
		this.header = header;
	}
	public Date getSendDateTime() {
		return sendDateTime;
	}
	public void setSendDateTime(Date sendDateTime) {
		this.sendDateTime = sendDateTime;
	}
	public int getValidityTime() {
		return validityTime;
	}
	public void setValidityTime(int validityTime) {
		this.validityTime = validityTime;
	}
}
